storage/engine/rocksdb.go

```diff
@@ -663,9 +672,19 @@ func (r *rocksDBBatch) Commit() error {
 	}
 	C.DBBatchDestroy(r.batch)
 	r.batch = nil
+
+	// On success, run the deferred functions in reverse order.
+	for i := range r.defers {
+		r.defers[len(r.defers)-i-1]()
```

Comment:
I don't think we need to, but we should for consistency (plus otherwise if we leak the batch, we also leak the defers)